src/journal/journalctl.c

```diff
@@ -264,7 +264,11 @@ static int parse_boot_descriptor(const char *x, sd_id128_t *boot_id, int *offset
         sd_id128_t id = SD_ID128_NULL;
         int off = 0, r;
 
-        if (strlen(x) >= 32) {
+        if (streq(x, "all")) {
+                *boot_id = id;
```

Comment:
Eh, just say `*boot_id = SD_ID128_NULL; *offset = 0;`, that much more obvious to the reader, and much less likely to be messed up by some subsequent refactoring.